### ISSUE TITLE
CONTRIBUTING.md is unclear about standard library

### ISSUE THREAD

**Issue by araujoms** [JuliaLang/julia#53200]
I just submitted a PR to change something in `LinearAlgebra` (#53186). To do that I followed the instructions [here](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#contributing-to-the-standard-library) about how to contribute to the standard library, which didn't work. @jishnub then kindly remarked that I should instead follow the instructions for contributing to the core functionality or base libraries, which did work.

Either the instructions are incorrect or `LinearAlgebra` is not part of the standard library, despite being in the `stdlib` folder. Either way I think the documentation should be updated.

**Comment by ViralBShah** [comment]
Based on your experience, do you have a set of instructions that you can provide in a PR?

**Comment by araujoms** [comment]
I wouldn't do a PR here since I don't understand what is going on. I just think it would be helpful to clarify the status of `LinearAlgebra` and what instructions apply to it. Apparently it is part of the standard library, but one should use instead the instructions in the section "contributing to the core functionality or base libraries".

**Comment by araujoms** [comment]
To make matters more confusing `LinearAlgebra` appears to be in a [separate repository](https://github.com/JuliaLang/LinearAlgebra.jl) at first sight, but on closer inspection one sees that it's a zombie repository.

**Comment by inkydragon** [comment]
> Can we delete the old LinearAlgebra repo?

Or add a deprecation note and archive it.

**Comment by vtjnash** [comment]
I was looking at it, and couldn't tell if we needed to migrate the issues from it here first or just archive it

**Comment by ViralBShah** [comment]
I think we can just migrate the 6 issues and delete it. There is nothing there.

Separately, we may want to try and separate out LinearAlgebra again. IIRC, github gave us for free xlarge runners on the JuliaLang org, and that may be sufficient to get LinearAlgebra tests running.

**Comment by KristofferC** [comment]
Yeah, we can delete it for now. It's just confusing having it exist at this point and I don't think there is much value in it, easier to recreate it from scratch at some later point.

**Comment by vtjnash** [comment]
Hid a side-conversation about LinearAlgebra having a separate repo, since the resolution was that we deleted it now, and it detracts from the main conversation here about docs
